amaranth/lib/wiring.py

```diff
@@ -1496,13 +1514,13 @@ def connect(m, *args, **kwargs):
                     f"shape {_format_shape(member_shape)} because the shape widths "
                     f"({Shape.cast(first_member_shape).width} and "
                     f"{Shape.cast(member_shape).width}) do not match")
-            if first_member_reset_as_const.value != member._reset_as_const.value:
+            if first_member_init_as_const.value != member._init_as_const.value:
                 raise ConnectionError(
-                    f"Cannot connect together the member {_format_path(first_path)} with reset "
-                    f"value {first_member_reset!r} and the member {_format_path(path)} with reset "
-                    f"value {member.reset} because the reset values do not match")
+                    f"Cannot connect together the member {_format_path(first_path)} with init "
+                    f"value {first_member_init!r} and the member {_format_path(path)} with init "
+                    f"value {member.init} because the init values do not match")
```

Comment:
```suggestion
                    f"Cannot connect together the member {_format_path(first_path)} with initial "
                    f"value {first_member_init!r} and the member {_format_path(path)} with initial "
                    f"value {member.init} because the initial values do not match")
```